src/higley_lab_to_nwb/extractors/processed_imaging_extractor.py

```diff
@@ -0,0 +1,169 @@
+from pathlib import Path
+from typing import Tuple, Literal
+import numpy as np
+from skimage.exposure import rescale_intensity
+from roiextractors.extraction_tools import PathType, DtypeType, get_package
+from roiextractors.imagingextractor import ImagingExtractor
+import h5py
+
+
+from datetime import datetime
+
+
+def _load_data_from_dual_imaging_configuration(file_path: str, process_type: str) -> np.ndarray:
+    """Load pixels matrix from MATLab file generated processing imaging data in dual imaging configuration."""
+
+    with h5py.File(file_path, "r") as file:
+        pixels_matrix = file[process_type][:]
+
+        num_rows, num_columns = file["mask"][:].shape
+
+    return np.nan_to_num(pixels_matrix), num_rows, num_columns
+
+
+def _load_data_from_1p_imaging_configuration(file_path: str, process_type: str) -> np.ndarray:
+    """Load pixels matrix from MATLab file generated processing imaging data in 1p-imaging only configuration."""
+
+    with h5py.File(file_path, "r") as file:
+        pixels_matrix = file["dFoF"][process_type][:]
+
+        num_rows = int(file["R"][:])
+        num_columns = int(file["C"][:])
+
+    return np.nan_to_num(pixels_matrix), num_rows, num_columns
+
+
+def _get_mask_from_dual_imaging_configuration(file_path: str) -> np.ndarray:
+    with h5py.File(file_path, "r") as file:
+        mask = file["mask"][:]
+    return mask
+
+
+class ProcessedImagingExtractor(ImagingExtractor):
+    """Specialized extractor for reading processed imaging data from a MATLab file."""
+
+    extractor_name = "ProcessedImagingExtractor"
+    is_writable = True
+    mode = "file"
+
+    def __init__(
+        self,
+        file_path: PathType,
+        sampling_frequency: float,
+        process_type: Literal["dff_final", "dff_blue", "dff_uv", "blue", "uv", "green"],
+    ) -> None:
+        """Create a ProcessedImagingExtractor instance from a MATLab file.
+
+        Parameters
+        ----------
+        file_path : PathType
+            Path to the MATLab file.
+        sampling_frequency : float
+            The frequency at which the frames were sampled, in Hz.
+        process_type : str
+            The type of processed imaging data to extract.
+            Accepted values: "dff_final", "dff_blue", "dff_uv", "blue", "uv",  "green".
+        Notes
+        -----
+        """
+        super().__init__()
+        self.file_path = Path(file_path)
+        self._sampling_frequency = sampling_frequency
+        self._times = None
+
+        self._channel_names = process_type
+
+        if process_type in ["dff_final", "dff_blue", "dff_uv"]:
+            self._pixels_matrix, self._num_rows, self._num_columns = _load_data_from_dual_imaging_configuration(
+                file_path=file_path, process_type=process_type
+            )
+            mask = _get_mask_from_dual_imaging_configuration(file_path=file_path)
+        elif process_type in ["blue", "uv", "green"]:
```

Comment:
I thought it would be sufficient specifying it as Literal:
https://github.com/catalystneuro/higley-lab-to-nwb/blob/10e3e4c1abe665b2eba3b76e3d0f6ae9eaff931c/src/higley_lab_to_nwb/extractors/processed_imaging_extractor.py#L53

Reply:
that doesn't mean that it involves a check for the input, it shows the expected values but if you then call with "anythingelse" it will break the code

Reply:
where is the check?